src/applications/lgy/coe/status/components/DocumentUploader/DocumentUploader.jsx

```diff
@@ -1,131 +1,76 @@
-
-import { ACTIONS } from '../../../shared/constants';
+import { submitToAPI } from './submit';
+import { addFile } from './addFile';
 import { DOCUMENT_TYPES, FILE_TYPES } from '../../constants';
-import {
-  isValidFileType,
-  isNotBlank,
-  validateIfDirty,
-} from '../../validations';
+import { isNotBlank, validateIfDirty } from '../../validations';
 import FileList from './FileList';
 
-const initialState = {
-  documentType: DOCUMENT_TYPES[0],
-  documentDescription: {
-    dirty: false,
-    value: '',
-  },
-  errorMessage: null,
-  files: [],
-};
-
-const {
-  DOC_TYPE,
-  DOC_DESC,
-  FILE_UPLOAD_SUCCESS,
-  FILE_UPLOAD_FAIL,
-  FILE_UPLOAD_PENDING,
-  FORM_SUBMIT_FAIL,
-  DELETE_FILE,
-} = ACTIONS;
-
-const reducer = (state, action) => {
-  switch (action.type) {
-    case DOC_TYPE:
-      return { ...state, documentType: action.documentType };
-    case DOC_DESC:
-      return { ...state, documentDescription: action.documentDescription };
-    case FILE_UPLOAD_SUCCESS:
-      return {
-        ...state,
-        files: [...state.files, action.file],
-        documentType: DOCUMENT_TYPES[0],
-        documentDescription: {
-          dirty: false,
-          value: '',
-        },
-        errorMessage: null,
-      };
-    case FILE_UPLOAD_PENDING:
-      return { ...state, errorMessage: null };
-    case FORM_SUBMIT_FAIL:
-    case FILE_UPLOAD_FAIL:
-      return { ...state, errorMessage: action.errorMessage };
-    case DELETE_FILE:
-      return { ...state, files: action.files };
-    default:
-      return state;
-  }
-};
-
 const DocumentUploader = () => {
-  const [state, dispatch] = useReducer(reducer, initialState);
-
-  const { documentType, documentDescription, errorMessage, files } = state;
+  const token = localStorage.getItem('csrfToken');
+  const [state, setState] = useState({
+    documentType: '',
+    documentDescription: '',
+    errorMessage: null,
+    files: [],
+    successMessage: false,
+    submissionPending: false,
+    token: localStorage.getItem('csrfToken'),
+    reader: new FileReader(),
+  });
 
-  const errorMsgClass = errorMessage ? 'vads-u-padding-left--1p5' : null;
+  const errorMsgClass = state.errorMessage ? 'vads-u-padding-left--1p5' : null;
   const disabledOnEmptyDescClass =
-    documentType === 'Other' && !isNotBlank(documentDescription.value)
+    state.documentType === 'Other' &&
+    !isNotBlank(state.documentDescription.value)
       ? 'file-input-disabled'
       : null;
 
-  const onSelectChange = useCallback(e => {
-    dispatch({ type: DOC_TYPE, documentType: e?.value });
-  }, []);
+  const onSelectChange = e => {
+    setState({ ...state, documentType: e.value });
+  };
 
-  const onTextInputValueChange = useCallback(e => {
-    dispatch({
-      type: DOC_DESC,
-      documentDescription: { dirty: true, value: e?.value },
+  const onTextInputValueChange = e => {
+    setState({
+      ...state,
+      documentDescriotion: e.value,
```

Comment:
*documentDiscription

Reply:
@jerekshoe Man my spelling is awful. Updated this